File: FrontEnd/lib/domain/interfaces/usecases/reminders/i_get_user_reminders_use_case.dart
/// @file i_get_user_reminders_use_case.dart
/// @description Interfaz del caso de uso para obtener los recordatorios del usuario.
/// @module Reminders
/// @layer Domain
library;

import '../../../entities/reminder.dart';

abstract interface class IGetUserRemindersUseCase {
  /// Devuelve los recordatorios activos del usuario autenticado.
  Future<List<Reminder>> execute();
}

File: BackEnd/src/domain/interfaces/usecases/reminders/IGetUserRemindersUseCase.ts
/**
 * @file IGetUserRemindersUseCase.ts
 * @description Interfaz del caso de uso para obtener los recordatorios del usuario.
 * @module Reminders
 * @layer Domain
 */

import type { ReminderResponseDTO } from '../../../dtos/reminders/reminder-response.dto.js';

export interface IGetUserRemindersUseCase {
  /**
   * @param userId — Id del usuario autenticado.
   * @returns Lista de recordatorios activos ordenados por scheduledDate asc.
   */
  execute(userId: string): Promise<ReminderResponseDTO[]>;
}
